src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -104,8 +104,14 @@ private String getParameter(HttpServletRequest request, String name, String defa
     return value == null ? defaultValue : value;
   }
 
-  /** Returns the ID of the current school. */
+  /** Returns the ID of the current school as an integer. */
   private int getId(HttpServletRequest request) {
-    return Integer.parseInt(request.getParameter("school-id"));
+    int fail = -1;
+    try {
+      return Integer.parseInt(request.getParameter("school-id"));
+    } catch (NumberFormatException exception) {
+      System.out.println("getID Invalid parametr: ID request is not a valid number.");
```

Comment:
Can we put a `TODO` comment here saying that in the future we should throw an exception and display an error message to the end users?